Documentation/DSCResources/DistributedSwitch/DistributedSwitch.md

```diff
@@ -0,0 +1,162 @@
+| **Server** | Key | string | Name of the Server we are trying to connect to. The Server can only be a vCenter. ||
+| **Credential** | Mandatory | PSCredential | Credentials needed for connection to the specified Server. ||
+| **Name** | Key | string | Name of the Distributed Switch located in the Datacenter specified in 'DatacenterName' key property. ||
+| **Location** | Key | string | Location of the Distributed Switch with name specified in 'Name' key property in the Datacenter specified in the 'DatacenterName' key property. Location consists of 0 or more Inventory Items. Empty Location means that the Distributed Switch is in the Network Folder of the Datacenter. The Root Folders of the Datacenter are not part of the Location. Inventory Item names in Location are separated by "/". Example Location for a Distributed Switch: "MySwitches/MyDistributedSwitches". ||
+| **DatacenterName** | Key | string | Name of the Datacenter we will use from the specified Inventory. ||
+| **DatacenterLocation** | Key | string | Location of the Datacenter we will use from the Inventory. Root Folder of the Inventory is not part of the Location. Empty Location means that the Datacenter is in the Root Folder of the Inventory. Folder names in Location are separated by "/". Example Location: "MyDatacentersFolder". ||
+| **Ensure** | Mandatory | Ensure | Value indicating if the Distributed Switch should be Present or Absent. | Present, Absent |
+| **ContactDetails** | Optional | string | The contact details of the vSphere Distributed Switch administrator. ||
+| **ContactName** | Optional | string | The name of the vSphere Distributed Switch administrator. ||
+| **LinkDiscoveryProtocol** | Optional | LinkDiscoveryProtocolProtocol | The discovery protocol type of the vSphere Distributed Switch that you want to create. If you do not set a value for this parameter, the default server setting is used. | CDP, LLDP, Unset |
+| **LinkDiscoveryProtocolOperation** | Optional | LinkDiscoveryProtocolOperation | The link discovery protocol operation for the vSphere Distributed Switch that you want to create. If you do not set a value for this parameter, the default server setting is used. | Advertise, Both, Listen, None, Unset |
+| **MaxPorts** | Optional | int | The maximum number of ports allowed on the vSphere Distributed Switch that you want to create. ||
+| **Mtu** | Optional | int | The maximum MTU size for the vSphere Distributed Switch that you want to create. Valid values are positive integers only. ||
+| **Notes** | Optional | string | The description for the vSphere Distributed Switch that you want to create. ||
+| **NumUplinkPorts** | Optional | int | The number of uplink ports on the vSphere Distributed Switch that you want to create. ||
+| **ReferenceVDSwitch** | Optional | string | The reference vSphere Distributed Switch. The properties of the new vSphere Distributed Switch will be cloned from the reference vSphere Distributed Switch. ||
```

Comment:
ReferenceVDSwitch in the resource is the name of the reference vdswitch. I think name is not enough to identify VDSwitch in the configuration. You can have multiple VDSSwitch instances with same name

Reply:
The VDSwitch Name is unique for each vCenter.